### PR TITLE
[WNMGDS-781] Rename dense table

### PR DESCRIPTION
**Pull request by nichia**
### Changed
- Rename `dense` to `compact` to match USWDS table

## How to test
- Check the Table component example to ensure that `dense` table is now referenced as `compact` table
- http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-781/rename-dense-table

### CODE REVIEW

File: packages/design-system/src/styles/components/_Table.scss
Comment (nichia) [inline packages/design-system/src/styles/components/_Table.scss:144]: @davidakennedy - the column header text instead of pseudo element fixes the issue of repeated information.

File: packages/design-system/src/components/Table/TableCell.jsx
Comment (bernardwang) [inline packages/design-system/src/components/Table/TableCell.jsx:75]: The problem with this is that this requires the HTML only examples to mainly add this code, which is quite alot of boilerplate

File: packages/design-system/src/styles/components/_Table.scss
Comment (bernardwang) [reply packages/design-system/src/styles/components/_Table.scss:144]: Same feedback [here](https://github.com/CMSgov/design-system/pull/927/files#r570382699). Lets hold off on addressing that issue in this ticket, we should file a new ticket for that and properly assess the different options before picking a solution

File: packages/design-system/src/types/Table/Table.d.ts
Comment (bernardwang) [inline packages/design-system/src/types/Table/Table.d.ts:21]: Similar to other props that are renamed, can we add a warning for folks using the `dense` prop? Search for `console.warn` for examples

File: packages/design-system/src/styles/components/_Table.scss
Comment (nichia) [reply packages/design-system/src/styles/components/_Table.scss]: Ok, have reverted back to using pseudo element. I've changed the <tr> display to `table-row`. Would that help address the `Firefox/NVDA` navigation issue?

Comment (nichia) [reply packages/design-system/src/styles/components/_Table.scss]: @davidakennedy = Here is an example of [responsive stacked table on hc.gov](https://www.healthcare.gov/third-party-privacy-policies/). The thead is `display: none` when stacked whereas we've just hidden the header display but still made it visible to screen reader. Do you think we should just make it `display: none`?

**Comment by davidakennedy** [comment]
I did a lot of testing on this today. @nichia and I have been talking about the two main issues I brought up in the last PR:

**Pseudo-elements getting announced**: Ultimately, I don't think this is a huge deal. But we could avoid it if we  did this:

```html
<td class="ds-u-text-align--left" role="cell" headers="column_1" data-title="Document title">
  <span aria-hidden="true">Declaration of Sentiments</span>
</td>
```

And putting the pseudo-element on the span. But after further testing, I don't think this is a must do. I'll leave it up to you all as to whether to implement it.

**Firefox/NVDA table navigation not working**: This is when you enter a table element, you should be able to press `CTL + ALT + arrow keys`. On small screen view, this breaks in Firefox. It does work with NDVA on Chrome and Edge. It seems to be related to the use of `display: block` on table elements. See [this bug for NVDA](https://github.com/nvaccess/nvda/issues/9879). This also occurs in the current live version of the table. I missed this when we first tested.

I tested [this example](https://codepen.io/aardrian/pen/LQddwO) from [this Adrian Roselli blog post](https://adrianroselli.com/2018/02/tables-css-display-properties-and-aria.html), which our code is based on, and that works as expected in Firefox/NVDA. So I'm curious what's different? @nichia will take a look.

It would be great if we can fix this. At least it's only affecting Firefox.

**Comment by bernardwang** [comment]
@davidakennedy @nichia I looked into the NVDA bug and the response table example from Adrian Roselli. 

From what I understand from the NVDA bug, setting the `<thead>` to `position: absolute` is what's causing NVDA to mess up, [according to this comment](https://github.com/nvaccess/nvda/issues/9879#issuecomment-576053393) it is the same as using `display: block`, which breaks table semantics. 

Interestingly, Adrian Roselli's table doesnt use `thead` at all, just a regular `<tr role="row">` containing `<th role="columnheader">`. When his table stacks at a small screensize, `display: none` is applied to the row that contains his column headers, so I believe the column is just not announced on a small screen size (unless you press the "add fake cell headers at narrow width" button). 

In our table, we originally used the `ds-u-visibility--screen-reader` utility to visually hide the `thead` element in order to preserve the native behavior of the column being announced. After reading these articles, I think we should try using `display: none` instead. This might fix issues with NVDA, and it would also fix the issue of the column header being repeated twice.

**Review by bernardwang — APPROVED** [review]
The `compact` renaming looks good to me! 

Approving, but the issues DK pointed out can be addressed under a different ticket and PR

**Comment by davidakennedy** [comment]
@nichia @bernardwang That's a good analysis. Bernard! Leaving this for the next iteration.

I tested this some more, and one key difference I noticed today after more testing, and what I think is leading to the NVDA table navigation not working:

- One small screen sizes, NVDA announces the table as having one column and one row. Which isn't true, but makes sense as to why NVDA won't work with the up, right or left arrow keys.
- If you navigate on Adrian Roselli's example, it announces the table correctly as having five columns, six rows. So that may be the issue.
- I do think going with `display: none` would be better for the double announcements.

**Comment by bernardwang** [comment]
>One small screen sizes, NVDA announces the table as having one column and one row. Which isn't true, but makes sense as to why NVDA won't work with the up, right or left arrow keys.

Oh hm, I wasnt aware that this was the case... that seems like a different issue altogether and I'm not sure what could be causing this.

**Comment by davidakennedy** [comment]
Something @bernardwang said made me look at the markup again. @nichia I think we need to put `role="rowgroup"` on any `thead`, `tbody` or `tfoot` elements. Because we're building a custom table after breaking the semantics, I think Firefox and NVDA see those as random `div`s inside a table. Untested, but I think that may help us here. See: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Table_Role

**Review by line47 — APPROVED** [review]

**Pull request merged**
